realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncConfigurationTests.java

```diff
@@ -116,6 +109,39 @@ public void serverUrl_setsFolderAndFileName() {
     }
 
     @Test
+    public void serverUrl_flexibleInput() {
+        // Check that the serverUrl accept a wide range of input
+        Object[][] fuzzyInput = {
+                // Only path -> Use auth server as basis for server url
+                { createTestUser("http://ros.realm.io/auth"), "/~/default", "realm://ros.realm.io/~/default" },
+                { createTestUser("http://ros.realm.io/auth"), "~/default", "realm://ros.realm.io/~/default" },
+                { createTestUser("https://ros.realm.io/auth"), "/~/default", "realms://ros.realm.io/~/default" },
+                { createTestUser("https://ros.realm.io/auth"), "~/default", "realms://ros.realm.io/~/default" },
+
+                // Forgot schema -> Use the one from the auth url
+                { createTestUser("http://ros.realm.io/auth"), "ros.realm.io/~/default", "realm://ros.realm.io/~/default" },
+                { createTestUser("http://ros.realm.io/auth"), "//ros.realm.io/~/default", "realm://ros.realm.io/~/default" },
+                { createTestUser("https://ros.realm.io/auth"), "ros.realm.io/~/default", "realms://ros.realm.io/~/default" },
+                { createTestUser("https://ros.realm.io/auth"), "//ros.realm.io/~/default", "realms://ros.realm.io/~/default" },
+
+                // Automatically replace http|https with realm|realms
+                { createTestUser(), "http://ros.realm.io/~/default", "realm://ros.realm.io/~/default" },
+                { createTestUser(), "https://ros.realm.io/~/default", "realms://ros.realm.io/~/default" }
+        };
+
```

Comment:
What about use cases where the user specify the [custom|default] authentication port will you skip the port?
Ex:
```java
 { createTestUser("http://ros.realm.io:7777/auth"), "ros.realm.io/~/default", "realm://ros.realm.io/~/default" }
```
?